autodiff/forward/forward.hpp

```diff
@@ -226,6 +226,13 @@ using common = typename std::common_type<A, B>::type;
 
 namespace traits {
 
+//-----------------------------------------------------------------------------
+// IS TYPE T A NUMBER?
+//-----------------------------------------------------------------------------
+template<typename T>
+struct isNumber : std::conditional_t<std::is_arithmetic_v<T>, std::true_type,
```

Comment:
What about we name this `isArithmetic` instead of `isNumber`? The reason is that I have been thinking about reserving the word `Number` as a generic term for autodiff types.

Reply:
I had same filings when did this PR. I’ll do it.